src/rendering/RenderUtil.cc

```diff
@@ -2058,6 +1869,259 @@ void RenderUtilPrivate::LowlightNode(const rendering::NodePtr &_node)
+        {
+          if (lSpotLight->Direction() !=
+            msgs::Convert(light.second.direction()))
+          {
+            lSpotLight->SetDirection(
+              msgs::Convert(light.second.direction()));
+          }
+        }
+        if (lSpotLight->InnerAngle() != light.second.spot_inner_angle())
+          lSpotLight->SetInnerAngle(light.second.spot_inner_angle());
+        if (lSpotLight->OuterAngle() != light.second.spot_outer_angle())
+          lSpotLight->SetOuterAngle(light.second.spot_outer_angle());
+        if (!ignition::math::equal(
+            lSpotLight->Falloff(),
+            static_cast<double>(light.second.spot_falloff())))
+        {
+          lSpotLight->SetFalloff(light.second.spot_falloff());
+        }
+      }
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+void RenderUtilPrivate::UpdateThermalCamera(const std::unordered_map<Entity,
+    std::tuple<double, components::TemperatureRangeInfo>> &_thermalCamData)
+{
+  for (const auto &thermal : _thermalCamData)
+  {
+    Entity id = thermal.first;
+    rendering::ThermalCameraPtr camera =
+        std::dynamic_pointer_cast<rendering::ThermalCamera>(
+        this->sceneManager.NodeById(id));
+    if (camera)
+    {
+      double resolution = std::get<0>(thermal.second);
+
+      if (resolution > 0.0)
+      {
+        camera->SetLinearResolution(resolution);
+      }
+      else
+      {
+        ignwarn << "Unable to set thermal camera temperature linear resolution."
+                << " Value must be greater than 0. Using the default value: "
+                << camera->LinearResolution() << ". " << std::endl;
+      }
+      double minTemp = std::get<1>(thermal.second).min.Kelvin();
+      double maxTemp = std::get<1>(thermal.second).max.Kelvin();
+      if (maxTemp >= minTemp)
+      {
+        camera->SetMinTemperature(minTemp);
+        camera->SetMaxTemperature(maxTemp);
+      }
+      else
+      {
+        ignwarn << "Unable to set thermal camera temperature range."
+                << "Max temperature must be greater or equal to min. "
+                << "Using the default values : [" << camera->MinTemperature()
+                << ", " << camera->MaxTemperature() << "]." << std::endl;
+      }
+    }
+  }
+}
+
+/////////////////////////////////////////////////
+void RenderUtilPrivate::UpdateAnimationWithRenderEngine(
+    const std::unordered_map<Entity, AnimationUpdateData> &_actorAnimationData)
+{
+  for (auto &it : _actorAnimationData)
+  {
+    auto actorMesh = this->sceneManager.ActorMeshById(it.first);
+    auto actorVisual = this->sceneManager.NodeById(it.first);
+    auto actorSkel = this->sceneManager.ActorSkeletonById(
+        it.first);
+    if (!actorMesh || !actorVisual || !actorSkel)
+    {
+      ignerr << "Actor with Entity ID '" << it.first << "'. not found. "
+             << "Skipping skeleton animation update." << std::endl;
+      continue;
+    }
+
+    const AnimationUpdateData &animData = it.second;
```

Comment:
Before moving this code to a helper method, `animData` was a non-const reference. However, based on what I could tell from how `animData` is being used, I believe that it can be a `const` reference. If anyone sees potential issues that could arise from this, let me know!